src/screens/TopicPageScreen/index.js

```diff
@@ -162,7 +162,7 @@ const TopicPageScreen = (props) => {
     try {
       const resultTopicDetail = await getTopics(domain);
       if (resultTopicDetail.data) {
-        const detail = resultTopicDetail.data[0];
+        const detail = resultTopicDetail.data.find((item) => item.name === domain);
         setTopicDetail(detail);
         handleFollowData(detail.is_followed_by);
         setMemberCount(Number(detail.followersCount));
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a potential issue. If `detail` is not found in the `resultTopicDetail.data` array, it will be `undefined`. This will cause an error when trying to access properties like `detail.is_followed_by` and `detail.followersCount`. To fix this, you should add a check to ensure `detail` is not `undefined` before using it.
```diff
-        const detail = resultTopicDetail.data.find((item) => item.name === domain);
-        setTopicDetail(detail);
-        handleFollowData(detail.is_followed_by);
-        setMemberCount(Number(detail.followersCount));
+        const detail = resultTopicDetail.data.find((item) => item.name === domain);
+        if (detail) {
+          setTopicDetail(detail);
+          handleFollowData(detail.is_followed_by);
+          setMemberCount(Number(detail.followersCount));
+        } else {
+          // Handle the case where detail is not found
+        }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->